src/util/key.rs

```diff
@@ -64,6 +64,32 @@ impl PublicKey {
     }
 }
 
+impl fmt::Display for PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.compressed {
+            for ch in &self.key.serialize()[..] {
+                write!(f, "{:02x}", ch)?;
+            }
+        } else {
+            for ch in &self.key.serialize_uncompressed()[..] {
+                write!(f, "{:02x}", ch)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for PublicKey {
+    type Err = encode::Error;
+    fn from_str(s: &str) -> Result<PublicKey, encode::Error> {
+        let key = secp256k1::PublicKey::from_str(s)?;
+        Ok(PublicKey {
+            key: key,
+            compressed: s.len() == 66
```

Comment:
This should be 34, right? Probably better to use the global constant in rust-secp.

Reply:
It should be 66. Where does 34 come from?

Reply:
I can use `2 * secp256k1::constants::PUBLIC_KEY_SIZE` if you'd like, but as I've said elsewhere, I don't like doing this because it forces the reader to go look that constant up to check that it's the correct one.

Reply:
Oooh, this is the hex length, not the byte length. Wow I got confused. I thought this was `65+1` somehow and you confused the bool (un<>compressed). My apologies :D 

Personally, if a constant is called `PUBLIC_KEY_SIZE`, and I'm depending on the library that exports this constant, I wouldn't check it.. In fact I think for most people that are not as thoroughly familiar with the matter, they, as I, would go check the constant to check if the magic constant is correct.